### DISCUSSION TITLE
Does the copilot retry menu exist for you in VSCode?

### DISCUSSION THREAD

**Discussion by FrostKiwi** [community/community#35623] [Copilot Conversations] [answered]
Can anyone using Visual Studio Code confirm whether or not the retry connection menu option is there or not when you click the copilot button?
![image](https://user-images.githubusercontent.com/60887273/193953581-926e61a9-002d-4041-ae12-60f2cbec3747.png)

I just wanna know whether it was patched out in a recent update or something else weird is happening with my instance of VSCode...

In reference to: https://github.com/community/community/discussions/35263 and https://github.com/microsoft/vscode/issues/162706#event-7521177604 being closed...

**Comment by igor-h** [comment, marked as answer]
Hi @FrostKiwi 

You need to pay to continue using it.
It was only a free demo

![image](https://user-images.githubusercontent.com/39650812/194900974-2941696e-85b8-49a5-95e5-ad3b4b527a4b.png)

LOG:
```
Invalid copilot token: missing token: 403
[ERROR] [default] [2022-10-06] GitHub Copilot could not connect to server. Extension activation failed: "Thank you for participating in the GitHub Copilot Technical Preview. Your extended access has ended.
```


[Billing github copilot](https://docs.github.com/en/billing/managing-billing-for-github-copilot/about-billing-for-github-copilot)

**GitHub Copilot pricing**
The GitHub Copilot subscription is available on a monthly or yearly cycle. If you choose a monthly billing cycle, you will be billed **_$10 per calendar month. If you choose a yearly billing cycle, you will be billed $100 per year._**

_If this was helpful mark this comment, as resolved_  :) 
Thanks!

**Reply by FrostKiwi** [reply]
> You need to pay to continue using it. It was only a free demo


@igor-h 
I'm still within my subscription period and CoPilot works when it manages a connection:
![image](https://user-images.githubusercontent.com/60887273/194966033-70646ee8-bc58-4d1f-92b4-b1c911298d0b.png)
My question is regarding the context menu in your screenshot. Does it still exist as of today? Was it patched out recently or is my VSCode install borked?

**Reply by igor-h** [reply]
Nop doesn't exist anymore for me

**Reply by FrostKiwi** [reply]
> Nop doesn't exist anymore for me

Thank you for confirming, that I have not gone insane :D
Now this is a major deal break, since now the retry option is not accessible anymore. No command, no nothing.
Now where the hell do you create an issue for that? https://github.com/microsoft/vscode/issues/162706#event-7521177604 was closed for being in the wrong repo and https://github.com/orgs/community/discussions/35263 is a community tab, not an issue tracker.

Anyone any ideas?

**Reply by igor-h** [reply]
@FrostKiwi  Try to use shortcuts to switch between results:
https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-visual-studio-code#keyboard-shortcuts-for-github-copilot
👍🏽 
(if with this you will resolve your doubts, I'll apreacite if you mark my answer as right) 
:)

**Reply by FrostKiwi** [reply]
@igor-h
as already mentioned, there exists no command or keyboard shortcut exposed by the Copilot extension, that allows you to reconnect. I already searched for the command, that the retry button is supposed to invoke.

This is really bizarre, like someone accidentally killed  a whole context menu by accident with a recent update. But there exists no issue tracker to report it. 

Unfortunately this doesn't solve my issue, because I still can't access the retry button or the command this retry button issues.

**Comment by FrostKiwi** [comment]
@igor-h 
> Nop doesn't exist anymore for me

Actually, now that I think about it. You totally answered the question of this post.
The retry context menu is gone. Thx.

**Reply by igor-h** [reply]
Thanks you!
